### PR TITLE
#13(feature):   signIn with email and password

### PR DESCRIPTION
**Pull request by faid-terence**
#### What does this PR do?
This pull request adds login functionality using email and password to the project. Users can now sign in to the app by entering their email and password.

####  Description of Task to be completed?

The vendor and the buyer should be able to sign in to app by providing their email & password.
The API should return a JW token on a successful login.
The administrator should be able to sign in to the app using pre-configured login credentials.
Appropriate error messages are returned on unsuccessful attempts
#### How should this be manually tested?
First Create an Account
Use registered credentials to SignIn
#### Any background context you want to provide?
Provided token expires in 1 hour

#### What are the relevant pivotal tracker/Trello stories?
Trello Card : https://trello.com/c/j2PkFzpM/13-sign-in-with-email-password

#### Questions:

### CODE REVIEW

**Comment by faid-terence** [comment]
[Sign in with Email & Password](https://trello.com/c/j2PkFzpM/13-sign-in-with-email-password)

File: coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chanceDira) [inline coverage/clover.xml]: This coverage folder should be in .gitignore

File: src/controller/authController.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+ export const UserLogin = async (req, res) => {
+  const { email, password } = req.body;
+
+  
+  try {
+    // Check if user with the given email exists
+    const user = await db.User.findOne({ where: { email } });
+    if (!user) {
+      return res.status(401).json({ message: "Invalid Credentials😥" });
```
Comment (chanceDira) [inline src/controller/authController.js]: Let's use Jsend format

File: src/database/seeders/20230403203144-ROLEs.cjs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        createdAt: new Date(),
+        updatedAt: new Date()
+      },
+      {
+        name: 'user',
+        createdAt: new Date(),
+        updatedAt: new Date()
+      },
+      {
+        name: 'vender',
```
Comment (chanceDira) [inline src/database/seeders/20230403203144-ROLEs.cjs]: vendor instead of vender

File: src/database/seeders/20230403203144-ROLEs.cjs [hunk 4]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up (queryInterface, Sequelize) {
+    await queryInterface.bulkInsert('ROLEs', [
```
Comment (chanceDira) [inline src/database/seeders/20230403203144-ROLEs.cjs]: Why roles is written in this case

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+          npm install
+          npm run build
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: npm install
```
Comment (ALVINdimpos) [inline .github/workflows/node.js.yml]: remember to add deploy

**Review by chanceDira — CHANGES_REQUESTED** [review]
Good job @faid-terence just some changes

File: .github/workflows/node.js.yml (hunk 5 above)
Comment (faid-terence) [reply .github/workflows/node.js.yml]: Done!

File: src/controller/authController.js (hunk 2 above)
Comment (faid-terence) [reply src/controller/authController.js]: Done!

File: coverage/clover.xml (hunk 1 above)
Comment (faid-terence) [reply coverage/clover.xml]: Done!

File: src/database/seeders/20230403203144-ROLEs.cjs (hunk 3 above)
Comment (faid-terence) [reply src/database/seeders/20230403203144-ROLEs.cjs]: Solved!

File: src/database/seeders/20230403203144-ROLEs.cjs (hunk 4 above)
Comment (faid-terence) [reply src/database/seeders/20230403203144-ROLEs.cjs]: Type erro!, Solved!

**Review by bryan936M — APPROVED** [review]
This is great progress!

**Review by yobuir — APPROVED** [review]
nice job @faid-terence

**Review by ALVINdimpos — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -12,13 +12,12 @@
     "migrate": "npx sequelize-cli db:migrate",
     "make-seed": "npx sequelize-cli seed:generate --name",
     "seed": "npx sequelize-cli db:seed:all",
-    "unseed": "npx sequelize-cli db:seed:undo",
-    "down": "npx sequelize-cli db:migrate:undo",
+    "unseed": "npx sequelize-cli db:seed:undo:all",
+    "down": "npx sequelize-cli db:migrate:undo:all",
     "test": "cross-env NODE_ENV=test jest",
-    "start": "cross-env NODE_ENV=production node src/server.js",
+    "start": "NODE_ENV=development nodemon --exec babel-node src/server.js",
```
Comment (Muhet) [inline package.json]: add cross env

File: src/controller/authController.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+export const UserLogin = async (req, res) => {
+  const { email, password } = req.body;
+
+  try {
+    // Check if user with the given email exists
+    const user = await db.User.findOne({ where: { email } });
+    
+    if (!user) {
+      return res
+        .status(401)
```
Comment (Muhet) [inline src/controller/authController.js:64]: const user = await db.User.findOne({ where: { email } });
if (!user) {
  return res.status(401).json({
    status: "fail",
    data: {
      message: "Invalid Credentials😥"
    }
  });
}


I think this is the good way you can use

**Review by Muhet — CHANGES_REQUESTED** [review]
try to learn and change them, But you did good job

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -1,4 +1,8 @@
+<<<<<<< HEAD
```
Comment (NTElissa) [inline .github/workflows/node.js.yml]: remove this conflict

**Review by NTElissa — COMMENTED** [review]
good progress but you have  to solve  and remove conflict  in deferent file  
check ( database file and seeders)

File: src/controller/authController.js (hunk 7 above)
Comment (faid-terence) [reply src/controller/authController.js:64]: Our TTL requested to use jsend format

**Review by NTElissa — APPROVED** [review]
RGTM

**Review by Muhet — APPROVED** [review]
good job

**Review by chanceDira — APPROVED** [review]
Nice work

**Pull request merged**